scripts/rolling_replace.py

```diff
@@ -0,0 +1,198 @@
+    parser.add_argument('--region', required=True,
+                        help='AWS region')
+    parser.add_argument('--batches', required=False, default=3,
+                        help='Number of batches in replacing instances')
+    parser.add_argument('--ami-id',
+                        help='AMI ID to verify in new instances.')
+    parser.add_argument('--force', '-f',
+                        help="Ignore downtime warning due to capacity.",
+                        default=False,
+                        action='store_true',
+                        )
+    return parser.parse_args()
+
+
+class RollingException(Exception):
+    pass
+
+
+class RollingTimeoutException(Exception):
+    pass
+
+
+def get_ami_id(instance):
+    for attr in instance.get('attributes'):
+        if attr.get('name') == 'ecs.ami-id':
+            return attr.get('value')
+    raise RollingException('No ami id found for this instance.')
+
+
+def get_services(ecs_client, cluster_name):
+    services = []
+    service_arns = ecs_client.list_services(
+        cluster=cluster_name
+    ).get('serviceArns')
+    for service_arn in service_arns:
+        services.append(service_arn.split('service/')[1])
+    return services
+
+
+def get_container_instance_arns(ecs_client, cluster_name):
+    instance_arns = []
+    next_token = ''
+    while True:
+        instances = ecs_client.list_container_instances(
+            cluster=cluster_name, maxResults=100, nextToken=next_token)
+        arns = instances.get('containerInstanceArns')
+        instance_arns += arns
+        next_token = instances.get('nextToken')
+        if not next_token:
+            break
+    return instance_arns
+
+
+def get_already_updated_instances(ecs_response, ami_id):
+    instances = []
+
+    for container_instance in ecs_response.get('containerInstances'):
+        instance_id = container_instance.get('ec2InstanceId')
+        status = container_instance.get('status')
+        if status == 'DRAINING':
+            # unexpected but we should proceed with terminating it
+            # because we already verified that the services were in a steady
+            # state.
+            utils.print_warning(f'{instance_id} was already draining')
+            continue
+        # batch_tasks_running += container_instance.get('runningTasksCount')
+        this_ami_id = get_ami_id(container_instance)
+        utils.print_info(f'Instance to drain: {instance_id}/{this_ami_id}')
+        if this_ami_id == ami_id:
+            utils.print_warning(
+                f'{instance_id} already uses ami_id {ami_id}. Skipping.'
+            )
+            instances.append(instance_id)
+    return instances
+
+
+def batch_instances(instances, batch_count):
+    batches = []
+    for i in range(0, len(instances), batch_count):
+        batches.append(instances[i:i + batch_count])
+    return batches
+
+
+def rolling_replace_instances(ecs, ec2, cluster_name, batches, ami_id, force):
```

Comment:
Does it make sense to split this into two functions? validate_replace_batch and rolling_replace?